src/runtime/builtin_modules/builtins.cpp

```diff
@@ -145,30 +145,68 @@ extern "C" Box* any(Box* container) {
     return boxBool(false);
 }
 
-extern "C" Box* min(Box* arg0, BoxedTuple* args) {
+Box* min_max(Box* arg0, BoxedTuple* args, BoxedDict* kwargs, int opid) {
     assert(args->cls == tuple_cls);
+    assert(kwargs->cls == dict_cls);
 
-    Box* minElement;
+    bool pass_kwargs = (kwargs && kwargs->d.size());
+    Box* key_func;
+    Box* extremElement;
     Box* container;
+    Box* extremVal;
+
+    if (pass_kwargs) {
+        static BoxedString* key_str = static_cast<BoxedString*>(PyString_InternFromString("key"));
+        auto it = kwargs->d.find(key_str);
+        key_func = it->second;
```

Comment:
we should have a check here that it was in fact "key" that was passed.  ex `min(1, a=1)` will segfault.  cpython checks both that the size of kwargs is 1 and that we found "key" in it.